content/articles/react_query-data-fetching-and-server-state-management/index.md

```diff
@@ -0,0 +1,207 @@
+ReactDOM.render(
+  <QueryClientProvider client={queryClient}>
+    <App />
+  </QueryClientProvider>,
+    document.getElementById('root')
+);
+```
+
+In the next steps, we will implement data fetching using the `useQuery`, `useMutation`, and the native React hooks.
+
+### Data fetching using React Query
+The difference between React Query and the common data fetching patterns such as `useEffect` is that React Query will first return the previously fetched data and then refetch it again. If the refetched resource is the same as the first, React Query will keep both as reference without forcing page reload which improves the application user experience.
+To demonstrate these concepts, we will use a `Post` and a `Posts` component. Go ahead and create a `components` folder in the root of our `src` application folder.
+
+To fetch data, we will include the logic in the `Posts.js` file:
+
+First, import React, `useQuery` hook, and `axios` library:
+
+```js
+import React from 'react'
+import {useQuery} from 'react-query'
+import axios from 'axios'
+```
+
+We need to define an asynchronous function `fetchPosts()` that will make network requests to the JSONPlaceholder API to fetch posts.
+
+```js
+
+async function fetchPosts(){
+    const {data} = await axios.get('https://jsonplaceholder.typicode.com/posts')    
+    return data
+}
+```
+
+Finally create a React functional component:
+
+```js
+
+function Posts(){
+    const {data, error, isError, isLoading } = useQuery('posts', fetchPosts) // first argument is a string to cache and track the query result
+    if(isLoading){
+        return <div>Loading...</div>
+    }
+    if(isError){
+        return <div>Error! {error.message}</div>
+    }
+
+    return(
+        <div className='container'>
+        <h1>Posts</h1>
+        {
+            data.map((post, index) => {
+                return <li key={index}>{post.title}</li>
+            })
+        }
+
+        </div>
+    )
+}
+
+export default Posts
+```
+
+Let's briefly dissect the above code:
+
+- We create a React functional component named `Posts` which is exported as default. 
+  
+- The  `useQuery` hook returns an application status which can be `isLoading` or `isError`, data and an error.
+  
+- `useQuery` hook takes two arguments  where the first one is a string to cache and track the query result. The second argument is the function we defined to make HTTP requests, `fetchPosts`. The react-query internally handles caching and updating functionality.
+- Before the return statement of the component, we execute a basic logic to check if the application is in loading state or if an error occured.
+- Inside the JSX part, we return the posts by mapping through the posts array.
+
+>>> For an awesome developer experience, react-query includes the in built react-query-devtools, so that we can be able to view the state and the cache. To enable in your application, ` import { ReactQueryDevtools } from 'react-query/devtools'`. 
```

Comment:
```suggestion
> For an awesome developer experience, react-query includes the in built react-query-devtools, so that we can be able to view the state and the cache. To enable in your application, ` import { ReactQueryDevtools } from 'react-query/devtools'`. 
```